src/ucp/core/ucp_worker.h

```diff
@@ -168,7 +169,10 @@ typedef struct ucp_worker {
     ucs_list_link_t               stream_eps;    /* List of EPs with received stream data */
     khash_t(ucp_worker_ep_hash)   ep_hash;       /* Hash table of all endpoints */
     khash_t(ucp_ep_errh_hash)     ep_errh_hash;  /* Hash table of error handlers associated with endpoints */
-    ucp_worker_iface_t            *ifaces;       /* Array of interfaces, one for each resource */
+    ucs_list_link_t               ifaces;        /* List of all interfaces (device and sockaddr),
+                                                    one for each resource */
+    ucp_worker_iface_t            **dev_ifaces;  /* Array of pointers to device interfaces,
```

Comment:
why can't you keep it as array of elements? And each element would be a list member if there is need

Reply:
@brminich So that if we need to remove an wiface of a listener it can be easily removed from a list without messing up he the indexes of the array.

Reply:
I mean you may allocate space for all device ifaces (num_tls), but sockaddr ifaces will be allocated separately. 
I'm not sure mem locality is important here, but you'd avoid many code changes. Or am I missing something?

Reply:
You mean to keep a separate array for the sockaddr wifaces?
Since we don't know in advance how many of those there will be, as opposed to devices wifaces during worker_create, if the sockaddr wifaces are on an array, will need to realloc on every new one.

Reply:
ok then, I thought we always have `num_tls` device ifaces.

Reply:
We do have num_tls device ifaces and pointers to them will be kept in an array. But we will have an additional unknown number of sockaddr ifaces which won't be in the array.